src/main/bx/public/mcp.bxm

```diff
@@ -0,0 +1,65 @@
+<bx:output><!---
+/**
+ * [BoxLang]
+ *
+ * Copyright [2023] [Ortus Solutions, Corp]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ * ----------------------------------------------------------------------------------
+ * MCP Server Entry Point (Front Controller)
+ *
+ * This file serves as the HTTP front controller for the Model Context Protocol (MCP) server.
+ * It receives and validates requests, then delegates processing to MCPRequestProcessor.
+ *
+ * Usage:
+ * 1. Register your tools in Application.bx or a startup script:
+ *    mcpServer( "default" ).registerTool( aiTool( "myTool", "Description", handler ) );
+ *
+ * 2. Access this endpoint:
+ *    POST http://localhost/bxai/mcp.bxm
+ *
+ * 3. Send JSON-RPC requests:
+ *    {
+ *        "jsonrpc": "2.0",
+ *        "method": "tools/list",
+ *        "id": "1"
+ *    }
+ */
+--->
+<!--- Get request parameters --->
+<bx:set serverName = url.server ?: "default">
+<bx:set requestMethod = cgi.REQUEST_METHOD>
+<bx:set requestBody = requestMethod eq "POST" ? getHTTPRequestData().content : "">
+<bx:set acceptHeader = cgi.HTTP_ACCEPT ?: "application/json">
+
+<!--- Delegate to the request processor --->
+<bx:set processor = new bxModules.bxai.models.mcp.MCPRequestProcessor()>
+<bx:set result = processor.process(
+	serverName: serverName,
+	requestMethod: requestMethod,
+	requestBody: requestBody,
+	acceptHeader: acceptHeader,
+	urlParams: url
+)>
+
+<!--- Set response headers --->
+<bx:header name="Content-Type" value="#result.contentType#">
+<bx:loop collection="#result.headers#" item="headerValue" index="headerName">
+	<bx:header name="#headerName#" value="#headerValue#">
+</bx:loop>
+
+<!--- Set status code if not 200 --->
+<bx:if result.statusCode neq 200>
+	<bx:header statusCode="#result.statusCode#" statusText="#result.statusText#">
+</bx:if>
+
+<!--- Output response content --->
+#result.content#
```

Comment:
let's use `bx:content` component instead

```
bx:content type="application/pdf" variable="#myPdfVariable#" reset="true";
bx:content type="application/json" variable="#results.content# reset=true;
```

This way, it resets all buffers and we can set the content type immediately